conmon-rs/common/proto/conmon.capnp

```diff
@@ -5,7 +5,8 @@ interface Conmon {
     # Version
     struct VersionRequest {
         verbose @0 :Bool;
-        metadata @1 :Data;
+        metadataOld @1 :Data; # deprecated
```

Comment:
I believe it defeats the purpose of keeping the old metadata around if we change the name. Either we need to make a new name for the Metadata or just drop this value and claim incompatibilty with older clients.

Reply:
The name of a field is only relevant for the generated `proto` code. The binary encoding of capnp messages only depends on the field index `@1` (that didn't change).

Therefore the binary protocol stays compatible with older versions of the protocol. Only the library code that fills the generated `proto` structs with data needs to change, but that change is not user facing and therefore also not a breaking change.

Reply:
ah so if I have an old client and I'm accessing @1, I will just name it `metadata` and have no idea the server conceives of it as `metadataOld`?

Reply:
Yes, that's how it is.